Core/clim-core/table-formatting.lisp

```diff
@@ -373,60 +435,69 @@ skips intervening non-table output record structures."))
+                              initial-spacing (row-wise t))
   (setq x-spacing (parse-space stream (or x-spacing #\Space) :horizontal))
   (setq y-spacing (parse-space stream (or y-spacing
                                           (stream-vertical-spacing stream))
                                :vertical))
-  (with-new-output-record
-      (stream record-type item-list
-              :x-spacing x-spacing
-              :y-spacing y-spacing
-              :initial-spacing initial-spacing
-              :row-wise row-wise
-              :n-rows n-rows
-              :n-columns n-columns
-              :max-width max-width
-              :max-height max-height)
-    (multiple-value-bind (cursor-old-x cursor-old-y)
-        (stream-cursor-position stream)
-      (with-output-recording-options (stream :record t :draw nil)
-        (funcall continuation stream)
-        (force-output stream))
-      (adjust-item-list-cells item-list stream)
-      (setf (output-record-position item-list)
-            (stream-cursor-position stream))
-      (if move-cursor
-          (setf (stream-cursor-position stream)
-                (values (bounding-rectangle-max-x item-list)
-                        (bounding-rectangle-max-y item-list)))
-          (setf (stream-cursor-position stream)
-                (values cursor-old-x cursor-old-y)))
-      (replay item-list stream)
-      item-list)))
-
-(defun format-items (items
-                     &rest args
-                     &key (stream *standard-output*)
-                       printer presentation-type
-                       cell-align-x cell-align-y
-                       &allow-other-keys)
+  (apply 'invoke-with-new-output-record stream
+         (lambda (stream item-list)
+           (multiple-value-bind (cursor-old-x cursor-old-y)
+               (stream-cursor-position stream)
+             (with-output-recording-options (stream :record t :draw nil)
+               (funcall continuation stream)
+               (force-output stream))
+             (adjust-item-list-cells item-list stream)
+             (setf (output-record-position item-list)
+                   (stream-cursor-position stream))
+             (setf (stream-cursor-position stream)
+                   (if move-cursor
+                       (values (bounding-rectangle-max-x item-list)
+                               (bounding-rectangle-max-y item-list))
+                       (values cursor-old-x cursor-old-y)))
+             (replay item-list stream)
+             item-list))
+         record-type
+         :x-spacing x-spacing
+         :y-spacing y-spacing
+         :n-columns n-columns
+         :n-rows n-rows
+         :max-width max-width
+         :max-height max-height
+         :initial-spacing initial-spacing
+         :row-wise row-wise
+         (alexandria:remove-from-plist
+          initargs :record-type :x-spacing :y-spacing :n-columns :n-rows
+                   :max-width :max-height :initial-spacing :row-wise)))
```

Comment:
Instead of removing intiargs, this should be

(with-keywords-removed (initargs (:record-type))
  (apply stream lambda record-type initargs))

because otherwise when some other method verifies, whether the argument was supplied (for whatever reason), it will always conclude that yes, it was.

Reply:
See previous comment. But in this case I would argue that the intention was to default `row-wise` to `t`. Leaving the others alone seems like the right thing to do.

Reply:
I've answered in the previous comment, closing this thread.